tests/backup/backup_test.go

```diff
@@ -86,6 +88,508 @@ var _ = Describe("{UserGroupManagement}", func() {
 })
 
+var _ = Describe("{BkpRstrDiffK8sVerSimultaneousDiffNS}", func() {
+
+	var (
+		appList                          = Inst().AppList
+		backupNames                      []string
+		sourceClusterAppsContexts        []*scheduler.Context    // Each Context is for one Namespace which corresponds to one App
+		destinationClusterAppsContexts   []*scheduler.Context    // Each Context is for one Namespace which corresponds to one App
+		backupContexts                   []*BackupRestoreContext // Each Context is for one backup
+		restoreContexts                  []*BackupRestoreContext // Each Context is for one restore
+		restoreLaterContexts             []*BackupRestoreContext // Each Context is for one restore-later
+		preRuleNameList                  []string
+		postRuleNameList                 []string
+		appContexts                      []*scheduler.Context // All app contexts (namespaces) in this loop of the scaling
+		sourceNamespaces                 []string
+		destinationNamespaces            []string
+		namespaceMapping                 map[string]string
+		clusterUid                       string
+		clusterStatus                    api.ClusterInfo_StatusInfo_Status
+		cloudCredName                    string
+		cloudCredUID                     string
+		backupLocationUID                string
+		bkpLocationName                  string
+		srcMaj, srcMin, destMaj, destMin int64
+	)
+
+	backupLocationMap := make(map[string]string)
+	labelSelectors := make(map[string]string)
+	sourceNamespaces = make([]string, 0)
+	destinationNamespaces = make([]string, 0)
+	namespaceMapping = make(map[string]string)
+	providers := getProviders()
+
+	JustBeforeEach(func() {
+
+		StartTorpedoTest("BkpRstrDiffK8sVerSimultaneousDiffNS", "Backup on NS(=yy), K8-version(=x) ;; Restore on NS=yy, K8s-version=x+?(=z) [success]; Simultaneously, Restore on K8s-version=z, NS=abc [partial success]", nil, 0)
+
+		log.InfoD("Verifying if the pre/post rules for the required apps are present in the AppParameters or not ")
+		for i := 0; i < len(appList); i++ {
+			if Contains(postRuleApp, appList[i]) {
+				if _, ok := portworx.AppParameters[appList[i]]["post"]; ok {
+					dash.VerifyFatal(ok, true, "Post Rule details mentioned for the apps")
+				}
+			}
+			if Contains(preRuleApp, appList[i]) {
+				if _, ok := portworx.AppParameters[appList[i]]["pre"]; ok {
+					dash.VerifyFatal(ok, true, "Pre Rule details mentioned for the apps")
+				}
+			}
+		}
+
+	})
+
+	It("Backup on NS(=yy), K8-version(=x) ;; Restore on NS=yy, K8s-version=x+?(=z) [success]; Simultaneously, Restore on K8s-version=z, NS=abc [partial success]", func() {
+
+		Step("Verify if 'test app spec'", func() {
+
+			log.InfoD("Allowed apps are %v", appsWithCRDsAndWebhooks)
+			for i := 0; i < len(appList); i++ {
+				contains := Contains(appsWithCRDsAndWebhooks, appList[i])
+				dash.VerifyFatal(contains, true,
+					fmt.Sprintf("check if app [%s] can be used as a dummy spec for *this* test", appList[i]))
+			}
```

Comment:
Why would you want to do this? If you want to check if the app that you need for this test is part of the `app_list` then you have to do

```
contains := Contains(appList, appsWithCRDsAndWebhooks[0])
```

Reply:
It's the same logic as check for pre or post rules, where, if multiple apps are used to run this test at the same time, you'll need to verify that all the apps are actually valid, and not just the first app. Right?